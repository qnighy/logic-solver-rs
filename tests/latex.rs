use insta::assert_snapshot;
use std::process::Command;

#[test]
fn test_latex1() {
    assert_snapshot!(run("A"));
}

#[test]
fn test_latex2() {
    assert_snapshot!(run("A → A"));
}

#[test]
fn test_latex3() {
    assert_snapshot!(run("A → B → A"));
}

#[test]
fn test_latex4() {
    assert_snapshot!(run("(A → B → A) ∧ (A ∨ B → B ∨ A)"));
}

#[test]
fn test_latex5() {
    assert_snapshot!(run("(A → B → A) ∧ (A ∨ B → B)"));
}

#[test]
fn test_latex6() {
    assert_snapshot!(run("⇒"));
}

#[test]
fn test_latex7() {
    assert_snapshot!(run(
        "(A → A → B) → (B → B → C) → (C → C → D) → (D → D → E) → (E → E → F) → A → F"
    ));
}

fn run(s: &str) -> String {
    let output = Command::new("cargo")
        .args(&["run", "--", "-e", s, "--latex"])
        .output()
        .unwrap();
    assert!(output.status.success());
    String::from_utf8(output.stdout).unwrap()
}
