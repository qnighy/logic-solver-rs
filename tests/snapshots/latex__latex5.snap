---
source: tests/latex.rs
expression: "run(\"(A → B → A) ∧ (A ∨ B → B)\")"

---
\documentclass[a4paper,twoside=false,DIV=16]{scrartcl}
\usepackage{bussproofs}
\usepackage{colortbl}
\usepackage{varwidth}
\usepackage{pifont}
\usepackage{xcolor}
\usepackage{listings}
\usepackage{xpatch}
\usepackage{tikz}

\usetikzlibrary{graphs}
\usetikzlibrary{graphdrawing}
\usetikzlibrary{decorations.pathmorphing}
\usegdlibrary{layered}

\newcommand{\cmark}{\textcolor{blue}{\ding{52}}}%
\newcommand{\xmark}{\textcolor{red}{\ding{56}}}%

\lstset{%
  basicstyle={\small},%
  identifierstyle={\small},%
  commentstyle={\small\itshape},%
  keywordstyle={\small\bfseries},%
  ndkeywordstyle={\small},%
  stringstyle={\small\ttfamily},
  frame={tb},
  breaklines=true,
  columns=[l]{fullflexible},%
  numbers=left,%
  xrightmargin=0em,%
  xleftmargin=3em,%
  numberstyle={\scriptsize},%
  stepnumber=1,
  numbersep=1em,%
  lineskip=-0.5ex%
}

\title{Logic solver result}

\makeatletter
% remove author, date, publishers
\xpatchcmd{\@maketitle}
  {%
    {%
      \usekomafont{author}{%
        \lineskip .5em%
        \begin{tabular}[t]{c}
          \@author
        \end{tabular}\par
      }%
    }%
    \vskip 1em%
    {\usekomafont{date}{\@date \par}}%
    \vskip \z@ \@plus 1em
    {\usekomafont{publishers}{\@publishers \par}}%
  }
  {}{}{\PatchFailed}
\makeatother

\begin{document}

\maketitle


\begin{center}
\begin{tabular}{|c|c|}
  \hline
  \cellcolor[gray]{0.8}
  \begin{minipage}{5cm}
    \centering
    Classical Logic
  \end{minipage}
  &
  \cellcolor[gray]{0.8}
  \begin{minipage}{5cm}
    \centering
    Intuitionistic Logic
  \end{minipage}
  \\\hline

  {
    \huge
    \xmark
    Not provable
  }
  &
  {
    \huge
    \xmark
    Not provable
  }
  \\\hline
\end{tabular}
\end{center}

\section*{Input}

$(A \to B \to A) \wedge (A \vee B \to B)$

\section*{Refutation}

Refutation in Kripke model:

\begin{tikzpicture}[layered layout, level distance=2cm, sibling distance=2cm, nodes={{circle, draw}}, acc/.style={{decorate,decoration=snake}}]
\node (w0) {$W_0$};
\end{tikzpicture}

\begin{tabular}{r|c}
& $W_0$
\\\hline\hline
A & 1
\\
B & 0
\\
B \to A & 1
\\
A \to B \to A & 1
\\
A \vee B & 1
\\
A \vee B \to B & 0
\\
(A \to B \to A) \wedge (A \vee B \to B) & 0
\end{tabular}

\end{document}
